### DISCUSSION TITLE
Apply background on colored text

### DISCUSSION THREAD

**Discussion by microo8** [charmbracelet/lipgloss#78] [Q&A] [answered]
When aplying a style with background, it doesn't fill some whitespace after a colored text.
How to do it right?

example:
```go
package main

import (
	"fmt"
	"strings"

	"github.com/charmbracelet/lipgloss"
)

var (
	box = lipgloss.NewStyle().
		Background(lipgloss.Color("237")).
		Padding(2)
	label = lipgloss.NewStyle().
		Foreground(lipgloss.Color("6"))
	text = lipgloss.NewStyle().
		Foreground(lipgloss.Color("9"))
)

func main() {
	var col1, col2 strings.Builder
	col1.WriteString(label.Render("foo:"))
	col1.WriteRune('\n')
	col1.WriteString(label.Render("bar:"))
	col2.WriteString(text.Render("1235"))
	col2.WriteRune('\n')
	col2.WriteString(text.Render("ABC"))
	fmt.Println(
		box.Render(
			lipgloss.JoinHorizontal(
				lipgloss.Bottom,
				col1.String(),
				lipgloss.NewStyle().PaddingLeft(5).Render(col2.String()),
			),
		),
	)
}

```

output:
![screenshot](https://i.postimg.cc/pVKv5b6v/20220407-10h28m00s-grim.png)

**Comment by bashbunni** [comment, marked as answer]
Are you hoping to see more of the darker background right of "1235" and "ABC"? 
If so, I would use Lipgloss' [`Padding`](https://pkg.go.dev/github.com/charmbracelet/lipgloss#Style.Padding) or [`PaddingRight`](https://pkg.go.dev/github.com/charmbracelet/lipgloss#Style.PaddingRight) functions to tailor the padding to your liking. 

Let me know if that makes sense. I'm happy to answer any questions that pop up!
